pkg/skaffold/errors/err_def.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package errors
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/skaffold/proto"
+)
+
+type Error interface {
+	Error() string
+	StatusCode() proto.StatusCode
+	Suggestions() []*proto.Suggestion
+	Unwrap() error
```

Comment:
I don't think we need `Unwrap()` on our interface; we just need to ensure that `ErrDef` needs to have an `Unwrap()` method.  It doesn't hurt to keep this if you think it's helpful, but I think most users will use `errors.Unwrap(xxx)`, and more likely `errors.Is()` and `errors.As()`.